db/memory/src/main/java/org/commonjava/indy/mem/data/MemoryStoreDataManager.java

```diff
@@ -205,12 +208,12 @@ private void recurseGroup( final Group master, final List<ArtifactStore> result,
                 if ( recurseGroups && type == StoreType.group )
                 {
                     // if we're here, we're definitely recursing groups...
-                    recurseGroup( (Group) stores.get( key ), result, seen, includeGroups, true );
+                    recurseGroup( (Group) stores.get( key ), result, seen, includeGroups, true, enabledOnly );
                 }
                 else
                 {
                     final ArtifactStore store = stores.get( key );
-                    if ( store != null )
+                    if ( store != null && !( store.isDisabled() && enabledOnly ) )
                     {
```

Comment:
Agreed. This seemed the dangerous point to me too, I just didn't have the time to create a reasonable test for it. I can look into it after my PTO if no one else will.